tokio/src/io/io/repeat.rs

```diff
@@ -47,12 +48,14 @@ impl AsyncRead for Repeat {
     fn poll_read(
         self: Pin<&mut Self>,
         _: &mut Context<'_>,
-        buf: &mut [u8],
+        buf: &mut dyn BufMut,
     ) -> Poll<io::Result<usize>> {
-        for byte in &mut *buf {
-            *byte = self.byte;
+        // TODO(perf): could use `ptr::write`
+        let n = buf.remaining_mut();
+        while buf.has_remaining_mut() {
```

Comment:
To avoid virtual dispatch cost we could separately count number of written bytes so loop could be `while written_len < n`